src/components/transactions/SortModal.tsx

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import Modal from 'react-native-modal';
+import { TouchableOpacity } from 'react-native';
+import {
+  CloseButtonContainer,
+  SortModalContainer,
+  Styles,
+  SortVerticalSpacing,
+  CenteredContainer,
+  PaddedScrollView,
+} from './styles';
+import {
+  BlueText,
+  Body1Text,
+  ButtonTextWhite,
+  H4CardNavTab,
+} from '../../../assets/Fonts';
+import RadioButton from '../common/RadioButton';
+import {
+  SortTransactionDispatch,
+  SortTransactionState,
+  SortVoucherDispatch,
+  SortVoucherState,
+} from '../../utils/transactionUtils';
+import { ButtonMagenta } from '../../../assets/Components';
+
+type SortModalProps = {
+  name: 'invoices' | 'vouchers';
+  isVisible: boolean;
+  setIsVisible: (visibility: boolean) => void;
+  sortState: SortTransactionState | SortVoucherState;
+  sortDispatch: SortTransactionDispatch | SortVoucherDispatch;
+  sortDescriptions: string[];
+};
+export default function SortModal({
+  name,
+  isVisible,
+  setIsVisible,
+  sortState,
+  sortDispatch,
+  sortDescriptions,
+}: SortModalProps) {
+  return (
+    <Modal
+      isVisible={isVisible}
+      coverScreen={false}
+      style={Styles.modal}
+      backdropTransitionOutTiming={0}
+    >
+      <SortModalContainer>
+        <CloseButtonContainer>
+          <TouchableOpacity
+            onPress={() => {
+              setIsVisible(false);
+              sortDispatch({ type: 'RESET_IN_PROGRESS' });
+            }}
+          >
+            <BlueText>
+              <Body1Text>Close</Body1Text>
+            </BlueText>
+          </TouchableOpacity>
+        </CloseButtonContainer>
+
+        <PaddedScrollView alwaysBounceVertical={false}>
+          <H4CardNavTab>{`Sort ${name} by`}</H4CardNavTab>
+          <SortVerticalSpacing />
+          <RadioButton
+            data={sortDescriptions}
+            selected={sortState.inProgressSortType}
+            setSelected={(index: number) => {
+              sortDispatch({ type: 'SORT_BY', option: index });
+            }}
+          />
+
+          <SortVerticalSpacing />
+          <CenteredContainer>
+            <ButtonMagenta
+              onPress={() => {
+                setIsVisible(false);
+                sortDispatch({ type: 'ON_SUBMIT' });
+              }}
+            >
+              <ButtonTextWhite>Apply</ButtonTextWhite>
+            </ButtonMagenta>
+          </CenteredContainer>
+        </PaddedScrollView>
+      </SortModalContainer>
+    </Modal>
+  );
+}
```

Comment:
Would it make sense to have a clear field option on the sort modal? Considering that there's likely no default ordering, I'm not sure what it'd revert back to. However, always having a highlighted sort button might give the impression that some invoices are being sorted/shown improperly.

Reply:
The default ordering is date descending (newest first). I don't mind putting in a clear button, though we should probably check the design with Danielle first, but I also don't think it should give the impression of being shown improperly. A sort is just a reordering of the invoices, after all?